client/layout/index.jsx

```diff
@@ -155,10 +156,16 @@ const Layout = createReactClass( {
 				'is-active': this.props.isLoading,
 			} );
 
+		const { primarySiteId, recentSiteIds } = this.props;
+
 		return (
 			<div className={ sectionClass }>
 				<DocumentHead />
 				<SitesListNotices />
+				{ primarySiteId && <QuerySites siteId={ primarySiteId } /> }
+				{ map( recentSiteIds, recentSiteId => (
+					<QuerySites key={ recentSiteId } siteId={ recentSiteId } />
+				) ) }
```

Comment:
Do you really need the recent sites, too? To optimize loading wp.com and clicking on "My Sites", the primary site should be enough. Or am I missing something?

Reply:
Yeah, I've talked with @apeatling and he wanted the recents as well.

Reply:
Then I would do two improvements:
- extract this query-for-multiple-sites code into a separate component
- query the sites only if we don't already have them in state. Otherwise, you'll add 5+ extra requests to every Calypso initial load.

Reply:
> extract this query-for-multiple-sites code into a separate component
What kind of component do you have in mind? Like `<QueryRecentAndPrimarySites />`? Would it live in `components`? Wouldn't it be better if `QuerySites` accepted multiple site ids?

> query the sites only if we don't already have them in state. Otherwise, you'll add 5+ extra requests to every Calypso initial load

What if we have them in state but they have been updated externally? Or do you think the `/me/sites` request will handle that case?

Reply:
> What if we have them in state but they have been updated externally?

Yes, the `/me/sites` request will take care of that a few seconds later, as it did until now.

> What kind of component do you have in mind?

I think that `<QuerySites recentAndPrimarySites />` would be very nice. The syntax is very similar to `<QuerySites allSites />`. And there could be another parameter to fetch only if we already don't have the data. It makes sense for all three possible options: `siteId`, `allSites` and `recentAndPrimarySites`.